sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/package.scala

```diff
@@ -201,4 +201,29 @@ package object util extends Logging {
   def truncatedString[T](seq: Seq[T], sep: String, maxFields: Int): String = {
     truncatedString(seq, "", sep, "", maxFields)
   }
+
+  val METADATA_COL_ATTR_KEY = "__metadata_col"
+
+  /**
+   * Hidden columns are a type of metadata column that are not propagated through subquery aliases,
+   * and are candidates during qualified star expansions.
```

Comment:
Can we update the comments? It seems "not propagated through subquery aliases" is not a difference anymore. The new difference is "hidden column can be propagated through project".